euphonic/powder.py

```diff
@@ -340,13 +345,11 @@ def _qpts_cart_to_frac(qpts: Quantity,
     """
     lattice = crystal.reciprocal_cell()
 
-    return np.linalg.solve(lattice.to(ureg('1/bohr')).magnitude.T,
-                           qpts.to(ureg('1/bohr')).magnitude.T
-                           ).T
+    return np.linalg.solve(lattice.T, qpts.T).T.to('dimensionless').magnitude
```

Comment:
What is the `to('dimensionless')` doing here? Consider comment

Reply:
It resolves the unit scaling factors and gives a proper dimensionless number. Consider the case that we calculate an aspect ratio `aspect = length / height`. If length is in m and height is in cm, `aspect.magnitude` might not give the  value we expect as `aspect` might be using units of `m/cm`.

Similarly here it is quite possible that lattice is in Ångstrom and qpts are in reciprocal Bohr, so the fractional coordinates are calculated in Bohr/Angstrom - which is not what we want!

I suppose this can be commented with something like `# Cast 'dimensionless' to ensure length units are cancelled properly`. Would that explain it well enough to one unfamiliar with the problem?

Reply:
Yeah, that would work. It's just the phrase "to dimensionless" at first glance seems either pointless or deeply rooted in magic.